Add support for map for hot/cold clues

deathbeam opened the pull request:
- Improve the WorldMapPointManager to handle multiple world points
better
- Create new interface for clues that will return multiple locations,
have update and reset methods for clues with undetermined location
- Implement new LocationsClueScroll and LocationClueScroll to
HotColdClue to add support for displaying all the points on world map
- Handle the LocationsClueScroll to add each WorldPoint to the world map
when found
- Move part of the HotCold logic to the HotColdClue
- Improve the ClueScrollWorldMapPoint to not create BufferedImage for each of it's instance

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

MESLewis commented on runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapPoint.java:
```diff
@@ -28,10 +28,12 @@
 import net.runelite.api.coords.WorldPoint;
 
 @Data
+@EqualsAndHashCode(of = "worldPoint")
```
Will this cause two WorldMapPoint's of the same location to return as equal? Not very familiar with @EqualsAndHashCode

deathbeam replied:
Yeah. Not sure if that is the default behaviour, but i just abused the HashSet to avoid checking if worldpoints are already set in the list or not.

deathbeam replied:
I can probably extend it to tooltip too, or idk what else should I consider there.

MESLewis commented:
I think adding the display image should be enough. If it's showing the same image at the same location it is effectively the same point.

deathbeam commented:
Changed the hashset (well, removed it again and reverted to arraylist) due to some problems with it. Now it checks for bool if it was already set, also tested and it is sort of working, should be working for hot/cold too if anyone can test
